client/my-sites/sharing/connections/connection.jsx

```diff
@@ -78,12 +79,35 @@ class SharingConnection extends Component {
 		};
 	}
 
+	componentWillMount() {
+		this.fetchKeyringUser();
+	}
+
+	componentWillReceiveProps( nextProps, prevProps ) {
+		if ( nextProps.siteId !== prevProps.siteId || nextProps.connection !== prevProps.connection ) {
```

Comment:
Maybe I don't understand that right, and sorry if I am, but while in [componentWillReceiveProps](https://facebook.github.io/react/docs/react-component.html#componentwillreceiveprops) `prevProps` are `this.props`, so `fetchKeyringUser` is actually fetching the data for `prevProps` not `nextProps` as it should be?